tests/ui/ambiguous_associated_items.rs

```diff
@@ -0,0 +1,11 @@
+use pyo3::prelude::*;
+
+#[pyclass(eq)]
+#[derive(PartialEq)]
+pub enum Items {
+    Error,
+    Output,
+    Target,
+}
+
```

Comment:
There is possibly a similar case for `#[derive(IntoPyObject)]`, should probably add this as a second test.

Reply:
Yep, there is the same problem for `IntoPyObject`/`IntoPyObjectRef`😢

Reply:
I have added a test for `#[derive(IntoPyobject)]`. I have not managed to trigger the bug for `#[derive(IntoPyObjectRef)]` or any `#[pyclass]` scenario other than those added in the test file. If anyone else can give a failing example, I'll add it.

Reply:
Looks good to me 👍. I could not find anything else to trigger this either. (I guess `IntoPyObjectRef` does not trigger because `Self` refers to the reference type.)